File: Array&Hasning/3.TwoSum/TwoSum.py
from typing import List


def TwoSum(nums : List[int], target : int) -> List[int]:
    targetPairs = {}
    for i in range(len(nums)):
        difference = target - nums[i]
        if difference in targetPairs:
            return [targetPairs[difference], i]
        targetPairs[nums[i]] = i
    return []


print(TwoSum([4,5,6],10))

File: Array&Hasning/3.TwoSum/TwoSum.cs
static int[] TwoSum(int[] nums, int target)
{
    Dictionary<int,int> targetPairs = [];
    for(int i = 0; i < nums.Length; i++)
    {
        int difference = target-nums[i]; 
        if(targetPairs.ContainsKey(difference))
            return [targetPairs[difference], i];
        targetPairs.Add(nums[i], i);
    }
    return null;
}

int[] result = TwoSum([4,5,6],10);
Console.WriteLine($"[{result[0]},{result[1]}]");
